packages/thirdweb/src/abi/lib/getFunctionSelector.ts

```diff
@@ -0,0 +1,23 @@
+import type { AbiFunction } from "abitype";
+import { getFunctionSignature } from "./getFunctionSignature.js";
+import { slice } from "viem/utils";
+import type { Hash } from "viem";
+import { keccak_256 } from "@noble/hashes/sha3";
+import { uint8ArrayToHex } from "../../utils/uint8-array.js";
+
+const hash = (value: string) =>
+  `0x${uint8ArrayToHex(keccak_256(value))}` as const;
+
+const FN_SELECTOR_CACHE = /*@__PURE__*/ new Map<string, Hash>();
+
+export function getFunctionSelector(fn: string | AbiFunction): Hash {
+  const normalizedFn = getFunctionSignature(fn);
+  // if we have already computed the function selector before, return it
+  if (FN_SELECTOR_CACHE.has(normalizedFn)) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return FN_SELECTOR_CACHE.get(normalizedFn)!;
+  }
+  const functionSelector = slice(hash(normalizedFn), 0, 4);
```

Comment:
Could you use `toFunctionSelector` here?

https://viem.sh/docs/utilities/toFunctionSelector#tofunctionselector

Reply:
I believe previously I was using a different hash function here - very well possible this can now just use that util function directly yes - will look into updating this

Reply:
had to update viem version but yes that's helpful!